pkg/healthcheck/healthcheck.go

```diff
@@ -1031,7 +1036,10 @@ func (hc *HealthChecker) allCategories() []category {
 						}
 						cert, err := hc.fetchCredsFromSecret(tapTLSSecretName)
 						if err != nil {
```

Comment:
TIOLI: How about checking against `kerrors.IsNotFound(err)` instead of `nil` (like you did in `upgrade.go`) to avoid swallowing an error that might be different than the secret not being there? (Like RBAC issues, etc).